#### Problem Statement:
"Towards Safer Social Platforms: A Machine Learning and NLP Approach to Cyberbullying 
Detection"

#### Abstract:
In the today’s digital era everyone is eager to express their thoughts, emotions and 
on-going happenings in life to the outer world on various social media platforms. 
Social media is a vulnerable platform which is exposed to possible weakness and 
threats. Cyberbullying is the serious problem that can harm the mental and physical 
well-being of people who are using social media on daily basis to share their 
thoughts.
In contrast to traditional bullying, cyberbullying can happen at any moment, making 
it challenging for victims to avoid its consequences. Studies show that those who are 
the targets of cyberbullying may experience suicidal thoughts as well as feelings of 
depression, anxiety, and loneliness. Constant exposure to negative comments and 
cyberbullying erodes a person's confidence and self-worth, making them question 
their worth and ability. In addition, Long-term emotional scars result from victims 
who lose trust and avoid social connections in person as well as virtually

In [1]:
# load the libraries
import numpy as np
import pandas as pd

import matplotlib.pyplot as pls
import seaborn as sns

import warnings
warnings.filterwarnings('ignore')

In [3]:
data = pd.read_csv('output1.csv')

In [5]:
data

content  annotation_label
0                                 Get fucking real dude.                 1
1       She is as dirty as they come  and that crook ...                 1
2       why did you fuck it up. I could do it all day...                 1
3       Dude they dont finish enclosing the fucking s...                 1
4       WTF are you talking about Men? No men thats n...                 1
...                                                  ...               ...
19996    I dont. But what is complaining about it goi...                 0
19997   Bahah  yeah i&;m totally just gonna&; get pis...                 0
19998       hahahahaha >:) im evil mwahahahahahahahahaha                 0
19999            What&;s something unique about Ohio? :)                 0
20000              Who is the biggest gossiper you know?                 0

[20001 rows x 2 columns]

##### Understanding the data
- content: Text data, containing various statements
- annotation_label: Integer labels associated with the content
- 0 Non bullying text
- 1 Bullying text

##### Understanding the data through the basic analysis

In [10]:
data.shape

(20001, 2)

- The dataset contains 20,001 entries with two columns

##### Renaming  annotation_label  to target

In [30]:
data.rename(columns={'annotation_label':'target'},inplace=True)

In [32]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20001 entries, 0 to 20000
Data columns (total 2 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   content  20001 non-null  object
 1   target   20001 non-null  int64 
dtypes: int64(1), object(1)
memory usage: 312.6+ KB


- The data has no null values.
- There is a object data which needs a vectorization i.e the embedding concept to convert into its numerical format.
- The DataFrame has a memory usage of approximately 312.6 KB.

###### Checking for data balancing , since the target is intended for classification

In [35]:
data['target'].value_counts()

target
0    12179
1     7822
Name: count, dtype: int64

##### DataCleaning 